### PR TITLE
Update voidtools to support custom url and credentials

### PR DESCRIPTION
**Pull request by Audionut**
Allow to use reverseproxy url and/or custom port.
Allow credentials support.

### CODE REVIEW

**Comment by Audionut** [comment]
I've broken something, because all of the icons for everything in the settings page from voidtools and below is now borked.

**Comment by Purfview** [comment]
This looks like a dirty hack {I just quick looked at commits}.
Maybe proper way it would be implementing this as a special button with it's own section!?

**Comment by Audionut** [comment]
Yes, it will have to be something like that. Trying to hack the search urls like that completely borked it.

**Comment by Purfview** [comment]
I looked at it, maybe it's OK to go this way.
How custom 'Voidtools URL' expected to look like for example?

**Comment by Audionut** [comment]
It should do a replacement `search_url = search_url.replace("{voidSearchURL}", voidURL);` so that the search url = `(voidURL)/?search=%search_string%` for instance.

Instead the result is `https://www.imdb.com/title/tt2660888/{voidSearchURL}/?search=Star+Trek+Beyond`

Maybe I need to correct the search url somewhere else, rather than just before the get request?

The other solution was probably to move the definition of voidURL to the top of the script, so I could properly define the url in the definition of the search_url, instead of doing the {voidSearchURL} text replacement, but it didn't seem appropriate to have a random definition up there.

**Comment by Purfview** [comment]
Just wait a bit, I'll check what is what after I'll get my coffee.

**Comment by Audionut** [comment]
Just as a matter of being informative and correct.

```
  if (search_url.includes("{voidSearchURL}")) {
    console.log("search urls:", search_url);
    search_url = search_url.replace("{voidSearchURL}", voidURL);
    console.log("search url after replacement:", search_url);
  }
```

```
search urls: {voidSearchURL}/?search=Star+Trek+Beyond
search url after replacement: http://xxxxx/?search=Star+Trek+Beyond
```

So that part works as expected, and the icons displayed on site (IMDB) represent true and correct results. In other words, the GET request is working.

Something about having `'searchUrl': '{voidSearchURL}/?search=%tt%',` causes everything else to do crazy things though.

Thanks for looking, enjoy that coffee first.

**Comment by Purfview** [comment]
Check the commit at test branch (I didn't tested anything) : https://github.com/Purfview/IMDb-Scout-Mod/tree/test

I think that would be cleaner. 
Btw, refactoring should go to another PR.

**Comment by Purfview** [comment]
Oh `let search_url = search_url;` should be removed there I think.

File: IMDb_Scout_Mod.user.js [hunk 1]
```diff
@@ -5505,29 +5505,29 @@ var other_sites = [
       'both': true},
   {   'name': 'Voidtools-ID',
       'icon': 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAAAclBMVEUAAACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOACWOAD/gAD7fQDZZgDXZACnQwDrcgD0eADvdQClQgCkQQD3ewC2TQD3egDcaADVYwCwSgA9Lo9lAAAAFnRSTlMA/j7oBfCzq6FQ+Ujt25Z8XjYz39db4uTxuwAAAJlJREFUGNNNj9kWgjAMREkpFGUT1Okitqz//4ualgfmJblz5mTJkoTKKVciOyU7v0+YrK9kMrpFg6WXKuX9jM2RM9C+ZkPt2J5CitbAlmzkExzPGwLWgg0CSMZGg66JMWCOCWVh2iEbX/8aZ/QHYAIFC+14y+3+iUekOxJ/D7tiNq6SJ1NfFlSUdVzB/OCHmKLBfNGtad5X/gFwbwt+xeZTxgAAAABJRU5ErkJggg==',
-      'searchUrl': 'http://localhost:8080/?search=%tt%',
+      'searchUrl': 'http://voidtools.replacement/?search=%tt%',
```
Comment (Audionut) [inline IMDb_Scout_Mod.user.js:5508]: Update searchURL to a unique entry that can easily be replaced later.

File: IMDb_Scout_Mod.user.js [hunk 2]
```diff
@@ -6283,6 +6283,14 @@ var icon_sites = icon_sites_main.concat(special_buttons);
 
 async function replaceSearchUrlParams(site, movie_id, movie_title, movie_title_orig, movie_year, series_id, season_id, episode_id, mPOSTsearch) {
   var search_url = ('mPOST' in site && !mPOSTsearch) ? site['mPOST'] : site['searchUrl'];
+
+  voidURL = GM_config.get("void_url").replace(/\/+$/, "");
```
Comment (Audionut) [inline IMDb_Scout_Mod.user.js:6287]: Define voidURL for replacement searchURL replacement.

File: IMDb_Scout_Mod.user.js [hunk 3]
```diff
@@ -6283,6 +6283,14 @@ var icon_sites = icon_sites_main.concat(special_buttons);
 
 async function replaceSearchUrlParams(site, movie_id, movie_title, movie_title_orig, movie_year, series_id, season_id, episode_id, mPOSTsearch) {
   var search_url = ('mPOST' in site && !mPOSTsearch) ? site['mPOST'] : site['searchUrl'];
+
+  voidURL = GM_config.get("void_url").replace(/\/+$/, "");
+
+  var search_url = ('mPOST' in site && !mPOSTsearch) ? site['mPOST'] : site['searchUrl'];
+  if (search_url.includes("http://voidtools.replacement")) {
```
Comment (Audionut) [inline IMDb_Scout_Mod.user.js:6290]: Replace the placeholder with the actual wanted url.

File: IMDb_Scout_Mod.user.js [hunk 4]
```diff
@@ -7142,12 +7150,24 @@ async function maybeAddLink(elem, site_name, search_url, site, scout_tick, movie
       "Referer": "https://dontorrent.equipment"
     };
   }
+
+  //set voidtool credentials
```
Comment (Audionut) [inline IMDb_Scout_Mod.user.js:7154]: Define credentials for the GET header.

File: IMDb_Scout_Mod.user.js [hunk 5]
```diff
@@ -7142,12 +7150,24 @@ async function maybeAddLink(elem, site_name, search_url, site, scout_tick, movie
...
   GM.xmlHttpRequest({
     method: 'GET',
-    headers: reqHeader,
-    timeout: parseInt(GM_config.get('timeout_ms')),
     url: search_url,
+    timeout: parseInt(GM_config.get('timeout_ms')),
+    headers: (function() {
+      const reqHeader = {}; // Initialize headers as an empty object
+      // Check if the search_url starts with the specified voidURL
+      if (search_url.startsWith(voidURL)) {
```
Comment (Audionut) [inline IMDb_Scout_Mod.user.js:7166]: If the search url is a match for our void url, we need to set the proper authorization header.

File: IMDb_Scout_Mod.user.js [hunk 6]
```diff
@@ -11042,7 +11062,10 @@ function countSites(task) {
       'emby_password': {'type': 'text'},
       'emby_debug': {'type': 'checkbox'},
       'milkie_authToken': {'type': 'text'},
-      'tnt_authToken': {'type': 'text'}
+      'tnt_authToken': {'type': 'text'},
```
Comment (Audionut) [inline IMDb_Scout_Mod.user.js:11065]: Dump the user facing options in `Authorization Tokens:`

File: IMDb_Scout_Mod.user.js (hunk 1 above)
Comment (Audionut) [reply IMDb_Scout_Mod.user.js:5508]: Moved away from port 8080, in case some future event uses it. `voidtools.replacement` should forever be unique.

**Pull request closed without merging**